runtime/src/main/java/org/corfudb/runtime/view/workflows/RemoveNode.java

```diff
@@ -0,0 +1,74 @@
+package org.corfudb.runtime.view.workflows;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.protocols.wireprotocol.orchestrator.CreateWorkflowResponse;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.clients.ManagementClient;
+import org.corfudb.runtime.exceptions.WorkflowException;
+import org.corfudb.runtime.view.Layout;
+
+import javax.annotation.Nonnull;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ *
+ * A workflow request that makes an orchestrator call to remove a node from
+ * the cluster.
+ *
+ * Created by Maithem on 1/19/18.
+ */
+@Slf4j
+public class RemoveNode extends WorkflowRequest {
+
+    protected final String endpointToRemove;
+
+    public RemoveNode(@Nonnull String endpointToRemove, @Nonnull CorfuRuntime runtime,
+                      @Nonnull int retry, @Nonnull Duration timeout,
+                      @Nonnull Duration pollPeriod) {
+        this.endpointToRemove = endpointToRemove;
+        this.runtime = runtime;
+        this.retry = retry;
+        this.timeout = timeout;
+        this.pollPeriod = pollPeriod;
+    }
+
+    @Override
+    protected UUID sendRequest(@Nonnull Layout layout) {
+        // Select the current tail node and send an add node request to the orchestrator
+        CreateWorkflowResponse resp = getOrchestrator(layout).removeNode(endpointToRemove);
+        log.info("sendRequest: requested to remove {} on orchestrator {}:{}, layout {}",
+                endpointToRemove, getOrchestrator(layout).getRouter().getHost(),
+                getOrchestrator(layout).getRouter().getPort(), layout);
+        return resp.getWorkflowId();
+    }
+
+    @Override
+    protected ManagementClient getOrchestrator(@Nonnull Layout layout) {
```

Comment:
I think we should consolidate this logic.

Reply:
Yes, it definitely doesn't belong here? Also why does it override WorkflowRequest, does RemoveNode have a specific implementation of getOrchestrator (odd?)

Reply:
Factored out the getOrchestrator method. In initially, I wanted to select an orchestrator on the tail (as an optimization, since reads will be serviced locally) if the operation was add node and otherwise select a non-tail for remove operations.